### PR TITLE
Adds Random Roll buttons for Queue Console

### PR DESCRIPTION
**Pull request by Cupax3**
## About The Pull Request
Adds "Let's Roll":
Picks randomly from the list of abnormalities in the possible abnormalities.
Adds "Fuck It Let's Roll":
Picks randomly from all abnormalities in the unlocked levels. Locks all queue consoles afterwards to prevent metagaming.
Adds "Hardcore Fuck It Let's Roll":
Picks randomly from all levels. Locks all queue consoles afterwards to prevent metagaming.
Refactors some code to their own procs, adds some helper procs to SSabnormality_queue

## Why It's Good For The Game
~~it isnt???~~
<details>
<summary>CHAOS CHAOS</summary>

https://www.youtube.com/watch?v=Z01Tsgwe2dQ

</details>

## Changelog
:cl:
add: The Abnormality Queue console can now do the Manager's job for them by randomly choosing new abnormalities!
refactor: Cleaner queue console code, more SSabnormality_queue helper procs.
admin: Admins can now use admeme magic (MC -> SSabnormality_queue -> call proc "AnnounceLock") to lock all queue consoles for whatever possible reason they could fathom.
/:cl:

### CODE REVIEW

**Comment by Cupax3** [comment]
I HATE MAX LINE LENGTH LINTING
I HATE MAX LINE LENGTH LINTING
I HATE MAX LINE LENGTH LINTING

**Comment by Cupax3** [comment]
THE LINTER GOD IS APPEASED PRAISE THE SUN

**Comment by Kitsunemitsu** [comment]
Going to test this for a bit in any case.

**Review by Kitsunemitsu — APPROVED** [review]
Works very well, if anything I'd change it from "Let's rock" to "We ball

**Comment by Cupax3** [comment]
hi i just saw that the subsystem never clears its own flag that the pick was forced its ineffectual tight now but could be a bug in the future so dont merge pls ill push a fix tomorrow

**Comment by Cupax3** [comment]
> hi i just saw that the subsystem never clears its own flag that the pick was forced its ineffectual tight now but could be a bug in the future so dont merge pls ill push a fix tomorrow

done

File: code/controllers/subsystem/abnormality_queue.dm [hunk 1]
```diff
@@ -23,6 +23,8 @@ SUBSYSTEM_DEF(abnormality_queue)
 	var/next_abno_spawn = INFINITY
 	/// Wait time for next abno spawn
 	var/next_abno_spawn_time = 4 MINUTES
+	/// Tracks if the current pick is forced
+	var/fucked_it_lets_rolled = FALSE
```
Comment (EgorDinamit) [inline code/controllers/subsystem/abnormality_queue.dm:27]: Is there any reason for this to exist?

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
There's no need for new variable to lock consoles;
Just make hardcore roll change the list of available abnos to only one, so it cannot be changed.

**Comment by SpaceJohn413** [comment]
**FUCK IT WE BALL**
**FUCK IT WE BALL**
**FUCK IT WE BALL**
**FUCK IT WE BALL**

**Comment by Cupax3** [comment]
> There's no need for new variable to lock consoles;
> Just make hardcore roll change the list of available abnos to only one, so it cannot be changed.

I consider locking consoles to be a feature - for admin usage or future content that might require it.
If you'd still like me to remove it, please reply.

**Comment by Cupax3** [comment]
Actually
locking is kinda required because without it, you could re-press the roll buttons.
The option you mention is also already implemented via the new helper proc SSabnormality.ClearChoices()
I see nothing in the additions that is truly pointless - even the flag on the subsystem itself is a useful piece of knowledge as "There are no picks" is not always interchangeable with "This was rerolled".

**Review by EgorDinamit — DISMISSED** [review]

**Comment by Cupax3** [comment]
found an incorrect indendation e

**Review by dsmith328 — APPROVED** [review]
You know what, I think this is the greatest fucking Litmus test for managers and I'm not gonna specify *why*

**Review by EgorDinamit — APPROVED** [review]

**Pull request merged**